mlir/test/python/dialects/transform_memref_ext.py

```diff
@@ -16,6 +16,54 @@ def run(f):
     return f
 
 
+@run
+def testMemRefAllocaToAllocOpCompact():
+    sequence = transform.SequenceOp(
+        transform.FailurePropagationMode.Propagate,
+        [],
+        transform.OperationType.get("memref.alloc"),
+    )
+    with InsertionPoint(sequence.body):
+        module = transform.CastOp(
+            transform.OperationType.get("builtin.module"), sequence.bodyTarget
+        )
+        alloca = transform.CastOp(
+            transform.OperationType.get("memref.alloca"), sequence.bodyTarget
+        )
+        memref.MemRefAllocaToGlobalOp(module, alloca)
+        transform.YieldOp()
```

Comment:
I understand it's a test, but types looks quite wrong here. We are casting `op<"memref.alloc">` to two different other concrete types. This cannot be valid and may break should we introduce some static "type compatibility" checks in the cast. Sequence can have multiple block arguments, let's just use that and give them the expected type immediately.

Reply:
Yep, much better, thanks! Fixed in latest revsion.